src/extensions/timeseries/TimeseriesViewNew/useTimeseriesModel.ts

```diff
@@ -1,229 +1,161 @@
-import { useContext, useRef } from "react"
-import { createCalculationPool, HitherContext } from "../../common/hither"
+import { useContext, useMemo } from "react"
+import { createCalculationPool, HitherContext, HitherInterface } from "../../common/hither"
+import useFetchCache from "../../common/useFetchCache"
 import { RecordingInfo } from "../../extensionInterface"
 import Mda from "./Mda"
 
 interface MdaInterface {
   value: (i1: number, i2: number) => number
 }
 
-class TimeseriesModelNew {
-  _dataSegments = new Map<string, MdaInterface>()
-  _requestDataSegmentHandlers: ((ds_factor: number, segment_num: number) => void)[] = []
-  _dataSegmentSetHandlers: ((ds_factor: number, t1: number, t2: number) => void)[] = []
-  _dataSegmentsRequested = new Map<string, boolean>()
-  constructor(private args: { samplerate: number, num_channels: number, num_timepoints: number, segment_size: number }) {
-  }
-  clear() {
-    this._dataSegments.clear()
-    this._dataSegmentsRequested.clear()
-  }
-  setDataSegment(ds_factor: number, segment_num: number, X: MdaInterface) {
-    const code = `${ds_factor}:${segment_num}`
-    this._dataSegments.set(code, X)
-    this._dataSegmentSetHandlers.forEach(handler => {
-      handler(ds_factor, this.args.segment_size * ds_factor * segment_num, this.args.segment_size * ds_factor * (segment_num + 1))
-    })
-  }
-  onRequestDataSegment(handler: (ds_factor: number, segment_num: number) => void) {
-    this._requestDataSegmentHandlers.push(handler)
-  }
-  onDataSegmentSet(handler: (ds_factor: number, t1: number, t2: number) => void) {
-    this._dataSegmentSetHandlers.push(handler)
+// it may be important to limit this to 1 at a time when using a filter
+const timeseriesCalculationPool = createCalculationPool({maxSimultaneous: 1, method: 'stack'})
+
+export type TimeseriesData = {
+  getChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => number[]
+  requestChannelData: (ch: number, t1: number, t2: number, ds_factor: number) => void
```

Comment:
I assume these are two separate functions, one for getting what the data is (`getChannelData`) and one for making a request for data that we expect doesn't exist yet (`requestChannelData`)--would it be possible to do it so that the data consumer doesn't need to care whether the data has been requested before, just ask to access it & have the request automatically sent if it hasn't yet been made?

Or am I misunderstanding the difference between these two functions?